### PR TITLE
Errors in decryption error handling.

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: SignalServiceKit/src/Messages/OWSMessageDecrypter.m [hunk 1]
```diff
@@ -509,7 +516,8 @@ - (void)decryptUnidentifiedSender:(SSKProtoEnvelope *)envelope
             successBlock(result, transaction);
         } @catch (NSException *exception) {
             dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-                [self processException:exception envelope:envelope];
+                SSKProtoEnvelope *envelopeForException = (newEnvelope ?: envelope);
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Messages/OWSMessageDecrypter.m]: Change 1: use the "new" exception when possible when processing errors.

File: SignalServiceKit/src/Messages/OWSMessageDecrypter.m [hunk 2]
```diff
@@ -527,44 +535,48 @@ - (void)processException:(NSException *)exception envelope:(SSKProtoEnvelope *)e
...
-            OWSProdErrorWEnvelope([OWSAnalyticsEvents messageManagerErrorUntrustedIdentityKeyException], envelope);
-            OWSFailDebug(@"Failed to trust identity on incoming message from: %@", envelopeAddress(envelope));
-            return;
-        } else {
-            OWSProdErrorWEnvelope([OWSAnalyticsEvents messageManagerErrorCorruptMessage], envelope);
-            if (envelope.source.length > 0) {
-                errorMessage = [TSErrorMessage corruptedMessageWithEnvelope:envelope withTransaction:transaction];
-            } else {
-                TSErrorMessage *errorMessage = [TSErrorMessage corruptedMessageInUnknownThread];
+        @try {
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Messages/OWSMessageDecrypter.m]: Change 2: catch (and log) all exceptions in the exception handling logic.

File: SignalServiceKit/src/Messages/OWSMessageDecrypter.m [hunk 3]
```diff
@@ -527,44 +535,48 @@ - (void)processException:(NSException *)exception envelope:(SSKProtoEnvelope *)e
...
+            TSErrorMessage *errorMessage;
+
+            if (envelope.source.length < 1) {
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Messages/OWSMessageDecrypter.m]: Change 3: check for "no source" envelope case _first_, not _last_.

**Comment by michaelkirk-signal** [comment]
superseded by https://github.com/signalapp/Signal-iOS/pull/4045

**Pull request closed without merging**